txnsync/txngroups.go

```diff
@@ -157,18 +155,20 @@ func decodeTransactionGroups(data []byte, compressionFormat byte, genesisID stri
 	return txnGroups, nil
 }
 
-func decompressTransactionGroupsBytes(data []byte) (decoded []byte, err error) {
-	zr, err := gzip.NewReader(bytes.NewBuffer(data))
-	defer zr.Close()
-	if err != nil {
-		return nil, fmt.Errorf("error gzip decompressing data: %w", err)
+func decompressTransactionGroupsBytes(data []byte, lenDecompressedBytes uint64) (decoded []byte, err error) {
+	compressionRatio := lenDecompressedBytes / uint64(len(data)) // data should have been compressed between 0 and 95%
+	if lenDecompressedBytes > maxEncodedTransactionGroupBytes || compressionRatio <= 0 || compressionRatio >= 20 {
```

Comment:
for completeness, we also need to verify the compression ratio while we compress the data. If we get >95%, we should send it uncompressed so that it won't fail in the decompression.
( this is more of a hypothetical issue - I don't believe we'll be that lucky )

Reply:
please also have the magic number 20 define as a const.. but it looks good otherwise.